android/java/org/chromium/chrome/browser/autofill/BraveAutofillBackgroundServiceImpl.java

```diff
@@ -0,0 +1,483 @@
+
+            String country =
+                    getAutofillValueForSave(
+                            fields, nodesMap, context.getResources().getString(R.string.country));
+
+            AutofillAddress autofillAddress =
+                    new AutofillAddress(context, AutofillProfile.builder().build());
+            AutofillProfile profile = autofillAddress.getProfile();
+            profile.setSource(Source.ACCOUNT);
+            profile.setInfo(FieldType.ADDRESS_HOME_COUNTRY, country);
+            profile.setInfo(FieldType.PHONE_HOME_WHOLE_NUMBER, phone);
+            profile.setInfo(FieldType.NAME_FULL, name);
+            profile.setInfo(FieldType.EMAIL_ADDRESS, email);
+            profile.setInfo(FieldType.COMPANY_NAME, company);
+            profile.setInfo(FieldType.ADDRESS_HOME_STREET_ADDRESS, address);
+            profile.setInfo(FieldType.ADDRESS_HOME_CITY, city);
+            profile.setInfo(FieldType.ADDRESS_HOME_ZIP, postalCode);
+            profile.setInfo(FieldType.ADDRESS_HOME_STATE, state);
+
+            profile.setGUID(PersonalDataManager.getInstance().setProfileToLocal(profile));
+            autofillAddress.updateAddress(profile);
+        }
+        mSaveCallback.onSuccess();
+    }
+
+    private String getAutofillValueForSave(
+            @NonNull Map<String, AutofillId> fields,
+            Map<AutofillId, ViewNode> nodesMap,
+            String key) {
+        String autofillData = "";
+        if (fields.containsKey(key)) {
+            AutofillId autofillId = fields.get(key);
+            ViewNode node = nodesMap.get(autofillId);
+            int autofillType = node.getAutofillType();
+            AutofillValue autofillValue = node.getAutofillValue();
+            if (autofillType == View.AUTOFILL_TYPE_TEXT) {
+                autofillData =
+                        (autofillValue != null) ? autofillValue.getTextValue().toString() : "";
+            }
+        }
+        return autofillData;
+    }
+
+    private void addAutofillableFields(
+            Context context,
+            @NonNull Map<String, AutofillId> fields,
+            @NonNull ViewNode node,
+            @Nullable Map<AutofillId, ViewNode> nodesMap) {
+
+        String hint = getHint(context, node);
+        if (hint != null) {
+            AutofillId id = node.getAutofillId();
+            if (!fields.containsKey(hint)) {
+                fields.put(hint, id);
+                if (nodesMap != null) nodesMap.put(id, node);
+            }
+        }
+        int childrenSize = node.getChildCount();
+        for (int i = 0; i < childrenSize; i++) {
+            addAutofillableFields(context, fields, node.getChildAt(i), nodesMap);
+        }
+    }
+
+    @Nullable
+    protected String getHint(Context context, @NonNull ViewNode node) {
+        // First try the explicit autofill hints...
+
+        String[] hints = node.getAutofillHints();
+        if (hints != null) {
+            // We're simple, we only care about the first hint
```

Comment:
Why only first hint btw?

Reply:
we are getting autofill hints from node from first hint as of now. I'll work on this with follow up tasks for autofill passwords.